core/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -71,19 +71,22 @@ protected void configure() {
      * the setting during startup.
      */
     public void registerSetting(Setting<?> setting) {
-        switch (setting.getScope()) {
-            case CLUSTER:
-                if (clusterSettings.containsKey(setting.getKey())) {
-                    throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
-                }
-                clusterSettings.put(setting.getKey(), setting);
-                break;
-            case INDEX:
-                if (indexSettings.containsKey(setting.getKey())) {
-                    throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
-                }
-                indexSettings.put(setting.getKey(), setting);
-                break;
+        if (setting.isFiltered()) {
+            if (settingsFilterPattern.contains(setting.getKey()) == false) {
+                registerSettingsFilter(setting.getKey());
+            }
+        }
+        if (setting.hasClusterScope()) {
+            if (clusterSettings.containsKey(setting.getKey())) {
+                throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
+            }
+            clusterSettings.put(setting.getKey(), setting);
+        }
+        if (setting.hasIndexScope()) {
+            if (indexSettings.containsKey(setting.getKey())) {
+                throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
+            }
+            indexSettings.put(setting.getKey(), setting);
```

Comment:
Ok. Then, I guess I can remove https://github.com/elastic/elasticsearch/pull/16629/files#diff-e169910cb3c505e0a12af08b76ef9b4fR133 because we won't never allow a Setting with at least one default Property...

```java
        if (properties.length == 0) {
            this.properties = EnumSet.of(Property.NodeScope);
        } else {
            this.properties = EnumSet.copyOf(Arrays.asList(properties));
        }
```